### PR TITLE
Move HCL config to main

### PR DESCRIPTION
**Pull request by faisal-memon**
Decouples HCL config from the pkg that runs the daemon. This is how the main SPIRE repo is organized.

### CODE REVIEW

File: cmd/spiffe-helper/config.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
+package main
```
Comment (faisal-memon) [inline cmd/spiffe-helper/config.go:1]: This file is mostly copied form pkg/sidecar/config.go.

File: cmd/spiffe-helper/config/config_test.go [hunk 2]
```diff
@@ -10,7 +11,7 @@ import (
 )
 
 func TestParseConfig(t *testing.T) {
-	c, err := ParseConfig("../../test/fixture/config/helper.conf")
+	c, err := ParseConfig("../../../test/fixture/config/helper.conf")
```
Comment (MarcosDY) [inline cmd/spiffe-helper/config/config_test.go:14]: mmm looks like this config is only used for this test, may we move all test files to a folder inside this package?

File: cmd/spiffe-helper/main.go [hunk 3]
```diff
@@ -35,10 +36,36 @@ func startSidecar(configPath string, exitWhenReady bool, log logrus.FieldLogger)
...
+		SvidKeyFileName:          config.SvidKeyFileName,
+		SvidBundleFileName:       config.SvidBundleFileName,
+	}
+
+	for _, jwtSvid := range config.JwtSvids {
+		sidecarConfig.JwtSvids = append(sidecarConfig.JwtSvids, sidecar.JwtConfig{
+			JWTAudience:     jwtSvid.JWTAudience,
+			JWTSvidFilename: jwtSvid.JWTSvidFilename,
+		})
 	}
```
Comment (MarcosDY) [inline cmd/spiffe-helper/main.go:67]: may we send hclconfig to sidecar.New? so we can unit test that we are parsing it the right way? 
if not, may we move all this code inside hclconfig? (to test it too)

File: cmd/spiffe-helper/config/config.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+		if jwtConfig.JWTSvidFilename == "" {
+			return errors.New("'jwt_file_name' is required in 'jwt_svids'")
+		}
+		if jwtConfig.JWTAudience == "" {
+			return errors.New("'jwt_audience' is required in 'jwt_svids'")
+		}
+	}
+
+	if c.AgentAddress == "" {
+		c.AgentAddress = os.Getenv("SPIRE_AGENT_ADDRESS")
```
[the remark this answers is not in the document]
Comment (faisal-memon) [reply cmd/spiffe-helper/config/config.go:136]: I think we should do as a separate PR so it shows in commit log. I added that issue to the 0.8.0 milestone.

File: cmd/spiffe-helper/config/config_test.go (hunk 2 above)
Comment (faisal-memon) [reply cmd/spiffe-helper/config/config_test.go]: Done

File: cmd/spiffe-helper/main.go (hunk 3 above)
Comment (faisal-memon) [reply cmd/spiffe-helper/main.go]: Moved the code inside config/ directory. Will add some unit tests for it.

Comment (faisal-memon) [reply cmd/spiffe-helper/main.go]: Unit tests added.

File: cmd/spiffe-helper/main.go [hunk 5]
```diff
@@ -8,7 +8,7 @@ import (
 	"os/signal"
 
 	"github.com/sirupsen/logrus"
-	hclconfig "github.com/spiffe/spiffe-helper/cmd/spiffe-helper/config"
+	config "github.com/spiffe/spiffe-helper/cmd/spiffe-helper/config"
```
Comment (MarcosDY) [inline cmd/spiffe-helper/main.go:11]: this name is not required since it is the same than the package

**Review by MarcosDY — COMMENTED** [review]
just a minor comment, and can you resolve DCO, so we can merge?

File: cmd/spiffe-helper/main.go (hunk 5 above)
Comment (faisal-memon) [reply cmd/spiffe-helper/main.go]: Fixed

**Review by MarcosDY — APPROVED** [review]

**Pull request merged**